lib/Image/Adapter/Imagick.php

```diff
@@ -298,6 +309,32 @@ public function save($path, $format = null, $quality = null)
         return $this;
     }
 
+    /**
+     * @param string $format
+     * @param \Imagick|null $i
+     * @param bool $checkNumberOfImages
+     * @return bool
+     */
+    protected function checkPreserveAnimation(string $format = "", \Imagick $i = null, bool $checkNumberOfImages = true){
+        if (!$this->isPreserveAnimation()) {
+            return false;
+        }
+
+        if (!$i) {
+            $i = $this->resource;
+        }
+
+        if ($i && $checkNumberOfImages && $i->getNumberImages() <= 1) {
+            return false;
+        }
+
+        if ($format && !in_array(strtolower($format), ["gif", "original", "auto"])) {
```

Comment:
The format `original` and `auto` is never passed to the adapter, this is handled already in the processor: 
https://github.com/pimcore/pimcore/blob/1314ea955e06c82c9613619cee9c7e4a00651f1a/models/Asset/Image/Thumbnail/Processor.php#L361
So I think we need to check this in `getContentOptimizedFormat()` and probably in other places as well.

Reply:
In the `save()` action the format `original` is checked as well, so thought it should be considered here as well. See https://github.com/pimcore/pimcore/blob/497ab20541387e3ed5081a5a6775c59eadc045ed/lib/Image/Adapter/Imagick.php#L219